src/pages/index.js

```diff
@@ -0,0 +1,278 @@
+const previewModalCloseBtn = previewModal.querySelector(".modal__close-btn");
+
+let selectedCard;
+let selectedCardId;
+//remove evt.target.classList.toggle("card__like-button_active");
+//TODO 1. check whether card is currently liked or not
+//const isLiked = ???
+//TODO 2. call the changeLikesStatus method, passing it the appropirate arguments
+//TODO 3. handle the response (.then and .catch)
+//TODO 4. in the .then, toggle active class
+
+// TODO - if the card is liked, set the active class on the card
+function handleLike(evt, cardId) {
+  const likeBtn = evt.target;
+  const isLiked = likeBtn.classList.contains("card__like-btn_liked");
+
+  api
+    .changeLikeStatus(cardId, isLiked)
+    .then((updatedCard) => {
+      likeBtn.classList.toggle("card__like-btn_liked", !isLiked);
+    })
+    .catch(console.error);
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameElement = cardElement.querySelector(".card__title");
+  const cardImageElement = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-button");
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  cardDeleteBtn.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data);
+  });
+
+  cardImageElement.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_is-opened");
+  document.addEventListener("keydown", handleEsc);
+  modal.addEventListener("click", handleOverlay);
+}
+
+function closeModal(modal) {
+  if (!modal) return;
+  modal.classList.remove("modal_is-opened");
+  document.removeEventListener("keydown", handleEsc);
+  modal.removeEventListener("click", handleOverlay);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  // TODO Change text content to "Saving..."
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {});
+  //TODO - call setButtonText instead
+  // submitBtn.textContent = "Save";
+  setButtonText(submitBtn, false);
```

Comment:
` setButtonText(submitBtn, false);` should be in `finally`

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.